File: app/src/main/java/com/botob/ulisten2/media/impl/FakeMedia.java
package com.botob.ulisten2.media.impl;

import android.os.Parcel;

import com.botob.ulisten2.media.AbstractMedia;
import com.botob.ulisten2.media.MediaApp;
import com.botob.ulisten2.notification.NotificationData;

/**
 * @author boris on 7/8/18.
 */
public class FakeMedia extends AbstractMedia {

    public FakeMedia(NotificationData notificationData) {
        super(notificationData);
    }

    public FakeMedia(final Parcel parcel) {
        super(parcel);
    }

    public FakeMedia(final String title, final String album, final String artist,
                     final long timestamp, final String packageName) {
        super(title, album, artist, timestamp, packageName);
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        out.writeString(MediaApp.FAKE.name());
        super.writeToParcel(out, flags);
    }

    @Override
    protected String fetchTitle(NotificationData notificationData) {
        throw new UnsupportedOperationException("It is just a fake.");
    }

    @Override
    protected String fetchAlbum(NotificationData notificationData) {
        throw new UnsupportedOperationException("It is just a fake.");
    }

    @Override
    protected String fetchArtist(NotificationData notificationData) {
        throw new UnsupportedOperationException("It is just a fake.");
    }
}

File: app/src/main/java/com/botob/ulisten2/media/impl/FakeMedia.kt
package com.botob.ulisten2.media.impl

import android.os.Parcel
import com.botob.ulisten2.media.AbstractMedia
import com.botob.ulisten2.media.MediaApp
import com.botob.ulisten2.notification.NotificationData

/**
 * @author boris on 7/8/18.
 */
class FakeMedia : AbstractMedia {
    constructor(notificationData: NotificationData?) : super(notificationData) {}
    constructor(parcel: Parcel) : super(parcel) {}
    constructor(title: String?, album: String?, artist: String?,
                timestamp: Long, packageName: String?) : super(title, album, artist, timestamp, packageName) {
    }

    override fun writeToParcel(out: Parcel, flags: Int) {
        out.writeString(MediaApp.FAKE.name)
        super.writeToParcel(out, flags)
    }

    override fun fetchTitle(notificationData: NotificationData?): String? {
        throw UnsupportedOperationException("It is just a fake.")
    }

    override fun fetchAlbum(notificationData: NotificationData?): String? {
        throw UnsupportedOperationException("It is just a fake.")
    }

    override fun fetchArtist(notificationData: NotificationData?): String? {
        throw UnsupportedOperationException("It is just a fake.")
    }
}